### PR TITLE
fix: lib/runtime: fix nested storage transactions, fixes kusama block 4939774

### PR DESCRIPTION
**Pull request by noot**
## Changes

- fix nested storage transactions, fixes kusama block 4939774
- update `TrieState` API to have `BeginStorageTransaction`, `CommitStorageTransaction`, `RollbackStorageTransaction`

## Tests

you may need to install git-lfs for this to work https://git-lfs.github.com/

```
go test ./lib/runtime/storage
go test ./lib/runtime/wasmer -run TestInstance_ExecuteBlock_KusamaRuntime_KusamaBlock4939774
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- related to #1134

### CODE REVIEW

**Review by RyRy79261 — COMMENTED** [review]
Looks good, though received a pathing issue when running the block test:
image

File: lib/runtime/storage/trie.go [hunk 1]
```diff
@@ -55,6 +56,22 @@ func (s *TrieState) Snapshot() {
 	_ = s.t.Snapshot()
 }
 
+// BeginStorageTransaction begins a new nested storage transaction which will either be committed or rolled back at a later time.
```
Comment (arijitAD) [inline lib/runtime/storage/trie.go:59]: Acquire lock when updating the trie.

**Review by arijitAD — APPROVED** [review]
@noot  Is it guaranteed that there will be only one ongoing transaction at any point?

**Comment by noot** [comment]
@RyRy79261 yeah I was trying to use git lfs for that file since it's quite large but doesn't look like it worked, I'll try to upload it somewhere else

**Review by RyRy79261 — APPROVED** [review]

**Pull request merged**